seerep_srv/seerep_core_fb/src/core_fb_pointcloud.cpp

```diff
@@ -51,47 +51,50 @@ boost::uuids::uuid CoreFbPointCloud::addData(const seerep::fb::PointCloud2& pcl)
   return dataForIndices.header.uuidData;
 }
 
-void CoreFbPointCloud::addBoundingBoxesLabeled(const seerep::fb::BoundingBoxesLabeledStamped& boundingBoxeslabeled)
+void CoreFbPointCloud::addLabel(const seerep::fb::DatasetUuidLabel& datasetUuidLabel)
 {
   boost::uuids::string_generator gen;
-  boost::uuids::uuid uuidMsg = gen(boundingBoxeslabeled.header()->uuid_msgs()->str());
-  boost::uuids::uuid uuidProject = gen(boundingBoxeslabeled.header()->uuid_project()->str());
+  boost::uuids::uuid uuidMsg = gen(datasetUuidLabel.datasetUuid()->str());
+  boost::uuids::uuid uuidProject = gen(datasetUuidLabel.projectUuid()->str());
 
   auto hdf5io = CoreFbGeneral::getHdf5(uuidProject, m_seerepCore, m_hdf5IoMap);
 
-  hdf5io->writePointCloudBoundingBoxLabeled(boost::lexical_cast<std::string>(uuidMsg), boundingBoxeslabeled.labels_bb());
+  hdf5io->writeLabelsFb(boost::lexical_cast<std::string>(uuidMsg), datasetUuidLabel.datasetUuid()->str(),
+                        datasetUuidLabel.labels());
 
-  std::unordered_map<std::string, std::vector<seerep_core_msgs::LabelWithInstance>> labelWithInstancePerCategory;
-  for (auto bbWithCategory : *boundingBoxeslabeled.labels_bb())
+  // TODO this is the same code as in core_fb_image::addLabel
```

Comment:
The hole method could be shared between data types. Handle it in this PR or create an Issue.